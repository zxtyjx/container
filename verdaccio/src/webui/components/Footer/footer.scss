@import '../../styles/variable';

.wrap {
  margin-top: auto;
  border-top: 1px solid $border-color;
  background: #f9f9f9;
}
footer.footer {
  display: flex;
  margin-top: 0;
  padding: 20px 0;
  font-size: 14px;
  height: 18px;
  color: #999;

  span {
    display: inline-block;
    height: 18px;
    vertical-align: middle;
  }

  .tooltip {
    align-items: center;
    margin-left: 5px;
    position: relative;
    margin-top: -2px;
    height: 20px;
    background: #d3dddd;
    padding: 1px 4px;
    border-radius: 2px;
    transform: scaleX(0);
    transition: transform 0.3s cubic-bezier(.12,.76,.14,.99);
    transform-origin: left;

    &:before {
      position: absolute;
      top: 3px;
      left: -7px;
      display: block;
      content: '';
      width: 7px;
      height: 15px;
      background: #d3dddd;
      background-image: linear-gradient(45deg, white, white 30%, transparent 30%), linear-gradient(-225deg, white, white 30%, transparent 30%);
    }


    img:not(:first-child) {
      margin-left: 8px;
    }
  }

  // Footer Hover
  &.showAuthorsGeographic .tooltip,
  &:hover .tooltip {
    display: flex;
    transform: scaleX(1);
  }

  .earth {
    margin-left: 8px;
    margin-right: 8px;
  }

  :global {
    .emoji {
      width: 18px;
      height: 18px;
    }
  }

  .right {
    margin-left: auto;
  }
  .logo {
    vertical-align: top;
    height: 18px;
    width: 20px;
  }
}
