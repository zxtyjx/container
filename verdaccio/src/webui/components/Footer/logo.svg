<svg width="100" height="100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><path d="M48 17.6L32.8 48H24L.4.8h15.2l12.8 25.6 4.4-8.8H48z" id="a"/><filter id="b" filterUnits="objectBoundingBox" height="140.3%" width="139.9%" y="-11.7%" x="-20%"><feGaussianBlur stdDeviation="2.5"/></filter><path d="M50.8 12H35.6L41.2.8h15.2L50.8 12z" id="c"/><filter id="d" filterUnits="objectBoundingBox" height="269.6%" width="191.3%" y="-49.1%" x="-45.7%"><feGaussianBlur stdDeviation="2.5"/></filter><path d="M32.8 48H24L.4.8h15.2l20.377 40.89L32.8 48z" id="e"/></defs><path fill="none" d="M-1-1h582v402H-1z"/><g><g stroke="null" fill-rule="evenodd" fill="none"><use transform="matrix(1.71429 0 0 1.71429 -37.027 -40.362)" x="22.366" y="28.311" xlink:href="#a" filter="url(#b)" fill="#000"/><use transform="matrix(1.71429 0 0 1.71429 -37.027 -40.362)" x="22.366" y="28.311" xlink:href="#a" fill="#405236"/><path stroke="#405236" d="M80.27 40.4H58.816L50 58.028 26.785 11.6H5.33l38.4 76.8h12.542l24-48z" stroke-width="2.4"/><use transform="matrix(1.71429 0 0 1.71429 -37.027 -40.362)" x="22.366" y="28.311" xlink:href="#c" filter="url(#d)" fill="#000"/><use transform="matrix(1.71429 0 0 1.71429 -37.027 -40.362)" x="22.366" y="28.311" xlink:href="#c" fill="#CD4000"/><path stroke="#CD4000" d="M87.128 26.686L94.671 11.6H73.215l-7.543 15.086h21.456z" stroke-width="2.4"/><g><use transform="matrix(1.71429 0 0 1.71429 -37.027 -40.362)" x="22.366" y="28.311" xlink:href="#e" fill="#4A5E3F"/><path stroke="#405236" d="M56.274 88.4l4.415-8.763L26.783 11.6H5.33l38.4 76.8h12.547-.002z" stroke-width="2.4"/></g><path stroke="#CD4000" stroke-linecap="square" stroke-width="2.4" d="M65.771 11.6h26.094m-32.95 6.857h26.092m-32.95 8.229H78.15"/></g></g></svg>