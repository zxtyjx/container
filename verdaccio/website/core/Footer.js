/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

const React = require('react');

class Footer extends React.Component {
  render() {
    return (
      <footer className="nav-footer" id="footer">
        <section className="sitemap">
          <a href={this.props.config.baseUrl} className="nav-home">
            <img
              src={this.props.config.baseUrl + this.props.config.footerIcon}
              alt={this.props.config.title}
              width="66"
              height="58"
            />
          </a>
          <div>
            <h5>Docs</h5>
            <a
              href={
                this.props.config.baseUrl +
                'docs/' +
                this.props.language +
                '/what-is-verdaccio.html'
              }>
              Getting Started
            </a>
            <a
              href={
                this.props.config.baseUrl +
                'docs/' +
                this.props.language +
                '/docker.html'
              }>
              Docker
            </a>
            <a
              href={
                this.props.config.baseUrl +
                'docs/' +
                this.props.language +
                '/configuration.html'
              }>
              Configuration
            </a>
          </div>
          <div>
            <h5>Community</h5>
            <a
              href={
                this.props.config.baseUrl + this.props.language + '/users.html'
              }>
              User Showcase
            </a>
            <a
              href="https://stackoverflow.com/search?q=verdaccio" target="_blank">
              Stack Overflow
            </a>
            <a href="https://gitter.im/verdaccio" target="_blank">Project Chat</a>
            <a href="https://twitter.com/verdaccio_npm" target="_blank">
              <img
                alt="Follow Verdaccio on Twitter"
                src="https://img.shields.io/twitter/follow/verdaccio_npm.svg?label=Follow+Verdaccio&style=social"
              />
            </a>
          </div>
          <div>
            <h5>More</h5>
            <a href="https://medium.com/verdaccio" target="_blank">Blog</a>
            <a href="https://github.com/verdaccio" target="_blank">GitHub</a>
            <a
              className="github-button"
              href={this.props.config.repoUrl}
              data-icon="octicon-star"
              data-count-href="/verdaccio/verdaccio/stargazers"
              data-show-count={true}
              data-count-aria-label="# stargazers on GitHub"
              aria-label="Star this project on GitHub">
              Star
            </a>
          </div>
        </section>
      </footer>
    );
  }
}

module.exports = Footer;
