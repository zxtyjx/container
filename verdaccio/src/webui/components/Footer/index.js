import React, {Component} from 'react';

import classes from './footer.scss';
import logo from './logo.svg';
import earth from './earth.svg';

// Vectors from Twitter Emoji (Open Source)
import brazilFlag from './flags/brazil-1f1e7-1f1f7.svg';
import chinaFlag from './flags/china-1f1e8-1f1f3.svg';
import indiaFlag from './flags/india-1f1ee-1f1f3.svg';
import nicaraguaFlag from './flags/nicaragua-1f1f3-1f1ee.svg';
import pakistanFlag from './flags/pakistan-1f1f5-1f1f0.svg';
import spainFlag from './flags/spain-1f1ea-1f1f8.svg';

export default class Footer extends Component {
  constructor(props) {
    super(props);
    this.handleEarthIconClick = this.handleEarthIconClick.bind(this);
    this.state = {
      showAuthorsGeographic: false
    };
  }

  handleEarthIconClick() {
    this.setState({
      showAuthorsGeographic: true
    });
  }

  render() {
    return (
      <div className={classes.wrap}>
        <footer
          className={`container ${classes.footer} ${this.state.showAuthorsGeographic && classes.showAuthorsGeographic}`}
        >
          <span>Made with&nbsp;</span>
          <span>‚ù§</span>
          <span>&nbsp;on</span>
          <img className={`${classes.earth} emoji`} src={earth} alt="Earth" onClick={this.handleEarthIconClick}/>
          <div className={classes.tooltip}>
            <img src={brazilFlag} alt="Brazil" title="Brazil" className="emoji"/>
            <img src={chinaFlag} alt="China" title="China" className="emoji"/>
            <img src={indiaFlag} alt="India" title="India" className="emoji"/>
            <img src={nicaraguaFlag} alt="Nicaragua" title="Nicaragua" className="emoji"/>
            <img src={pakistanFlag} alt="Pakistan" title="Pakistan" className="emoji"/>
            <img src={spainFlag} alt="Spain" title="Spain" className="emoji"/>
          </div>
          {/* Countries are order by alphabets */}

          <div className={classes.right}>
            Powered by&nbsp;
            { /* Can't switch to HTTPS due it hosted on GitHub Pages */ }
            <a href="http://www.verdaccio.org/">
              <img className={classes.logo} src={logo} alt="Verdaccio" title="Verdaccio"/>
            </a>
            &nbsp;/&nbsp;
            {__APP_VERSION__}
            </div>
        </footer>
      </div>
    );
  }
}
