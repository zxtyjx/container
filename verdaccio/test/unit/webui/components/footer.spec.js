
import React from 'react';
import { mount } from 'enzyme';

import Footer from '../../../../src/webui/components/Footer/index';

describe('<Footer /> component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<Footer />);
    });

    it('should load the inital state of Footer component', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should hover the earth in footer', () => {
        const earth = wrapper.find('img[alt="Earth"]');
        earth.simulate('click');
        expect(wrapper.state('showAuthorsGeographic')).toEqual(true);
    });
});