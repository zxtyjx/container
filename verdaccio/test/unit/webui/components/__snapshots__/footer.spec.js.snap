// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component should load the inital state of Footer component 1`] = `"<div class=\\"wrap\\"><footer class=\\"container footer false\\"><span>Made with&nbsp;</span><span>‚ù§</span><span>&nbsp;on</span><img class=\\"earth emoji\\" src=\\"[object Object]\\" alt=\\"Earth\\"><div class=\\"tooltip\\"><img src=\\"[object Object]\\" alt=\\"Brazil\\" title=\\"Brazil\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"China\\" title=\\"China\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"India\\" title=\\"India\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Nicaragua\\" title=\\"Nicaragua\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Pakistan\\" title=\\"Pakistan\\" class=\\"emoji\\"><img src=\\"[object Object]\\" alt=\\"Spain\\" title=\\"Spain\\" class=\\"emoji\\"></div><div class=\\"right\\">Powered by&nbsp;<a href=\\"http://www.verdaccio.org/\\"><img class=\\"logo\\" src=\\"[object Object]\\" alt=\\"Verdaccio\\" title=\\"Verdaccio\\"></a>&nbsp;/&nbsp;1.0.0</div></footer></div>"`;
